### PR TITLE
Convert to typescript and use tslint

### PR DESCRIPTION
**Pull request by emmatyping**
This converts things to using Typescript. I also added tslint which found a few things I thought were good, we can rip it out if we don't like it (it also is rather opinionated on things I don't like).

You can see this live at https://www.ocf.berkeley.edu/~ethanhs/www/

Also the files didn't diff well so here is a nicer diff of the ts file and js original: https://www.diffchecker.com/NRrJuy3K

### CODE REVIEW

File: src/script.ts [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+
+    // Only update hours with the results from this request after the page has finished loading
+    promiseGetHours.then((hours) => {
+        if (hours[0] !== null) {
+            hours = hours[0];
+            const start = document.getElementById("starthours");
+            const end = document.getElementById("endhours");
+
+            if (start == null) {
+                console.log("document has not start time");
```
Comment (abizer) [inline src/script.ts]: document has not?

File: package.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "ocf-labmap",
+  "version": "0.1.0",
```
Comment (dkess) [inline package.json:3]: Is it possible to get rid of this? If not, what are the conventions around it? Do we have to update it or can we just ignore it?

File: src/script.ts [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+  });
+}
+
+function getDesktopsInUse() {
+    return get("https://www.ocf.berkeley.edu/api/lab/desktops").then(
+        (response) => JSON.parse(response)["public_desktops_in_use"],
+    );
+}
+
+function getHoursToday() {
```
Comment (dkess) [inline src/script.ts]: Should we type these two functions?

**Review by dkess — APPROVED** [review]
Thanks so much for this! lgtm

File: package.json (hunk 2 above)
Comment (emmatyping) [reply package.json:3]: The version is required AFAIK, but we can just ignore it.

File: src/script.ts [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+            if (parent == null) {
+                console.log("polygon has no parent");
+            } else {
+                (parent.parentNode as Element).classList.add("bounce");
+            }
+        }
+    }
+}
+
+// Start the HTTP request to get today"s hours before the page finishes loading
```
Comment (abizer) [inline src/script.ts]: s/''/'/

**Review by abizer — APPROVED** [review]

**Comment by jvperrin** [comment]
This looks great to me, but I'd probably change some stuff inside ocfweb to export information more suited for this page (like not having to parse the time in here for example). I think it would just make this simpler and less brittle

**Comment by emmatyping** [comment]
@jvperrin oh certainly agreed, but I think those changes can be done in later PRs

**Comment by jvperrin** [comment]
Yeah definitely, nice work on this!

**Pull request merged**